TESTDATA_GUIDELINES.md

```diff
@@ -0,0 +1,277 @@
+
+You can also see we're prepending the messages with a sequence number. We do this because it makes it easier for us to
+verify whether we could retrieve all sent messages form the reference data, and whether the order of messages is
+maintained in storage.
+
+Lastly, you're notice the messages aren't a simple `Hi!`. This is what we call self-descriptive data and for our
+research at the NFI we vastly prefer this over realistic messages.
+
+### Consider using self-descriptive data
+
+Depending on the goal of your reference data, you might want to consider using self-descriptive data. This is data of
+which the content describes itself, for example a chat message `This is a message from Alice to Bob sent at 2024-10-01
+14:53 UTC`, or a layer in a layered image file containing `This is text on layer 1, the letters are red, and there is
+a second layer with blue text`.
+
+Self-descriptive data comes in very handy when you're doing (forensic) analysis of a file format, as it makes it easier
+to interpret the data and prevent mistakes.
+
+### Generating large amounts of realistic data
+
+If the goal of your reference data is not to analyze the data content itself, but rather to create a large amount of
+synthetic that looks like real, organic user data, obviously self-descriptive data is not for you.
+
+If you want to create large amounts of realistic user data (like thousands of chat messages that look real), use Puma
+and have an LLM create the conversations for you. How to do that is beyond the scope of this document, but current day
+LLMs are smart enough to understand the Puma API (just give it the python code and the LLM will understand) and ask it
+to generate the python code containing the number of messages you need, on topics you want.
+
+## 2. Creating the test data
+
+After you have your script describing the actions you will take, you can either use Puma to create the test data, or
+create it manually.
+
+When deciding whether to use Puma, consider the following:
+
+* Does Puma already support your application?
+* Will you create the reference data multiple times? (eg on different phones, OS versions, or app versions)
+* Are you creating a large amount of data (eg 100+ messages)?
+
+If the answer to any of these questions is yes, our experience is that Puma will save you time.
+
+If Puma doesn't support your app, and this is a one-time thing, and you're just doing a dozen or so user actions, we
+recommend just doing ti manually
+
+### Creating test data with Puma
+
+If your app is not supported in Puma, you will need to add support for it. To do so, read
+the [contributing documentation](CONTRIBUTING.md).
+
+After you have written the Puma code and your Puma script, you can now execute it. Be sure to properly prepare the
+device (enable USB debugging!) and install the application. It's a good idea to go into the app settings and grant all
```

Comment:
Is it possible to refer to a place where we explain how to enable USB debugging?